hmpps-prisoner-search-indexer/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/indexer/listeners/OffenderEventListener.kt

```diff
@@ -88,19 +89,7 @@ class OffenderEventListener(
         else -> log.warn("We received a message of event type {} which I really wasn't expecting", eventType)
       }
     } catch (olfe: OptimisticLockingFailureException) {
-      if (olfe.message?.contains("Cannot index a document due to seq_no+primary_term conflict") == true) {
-        // This is not an error and so we want to avoid exceptions being logged
-        val (message, _, messageAttributes) = fromJson<Message>(requestJson)
-        log.info("Detected a seq_no+primary_term conflict and trying again for message:\n{}", message)
-        offenderEventQueueService.requeueMessageWithDelay(
-          requestJson!!,
-          messageAttributes.eventType.Value,
-          delayInSeconds = 1,
-        )
-      } else {
-        log.error("processOffenderEvent() Unexpected error", olfe)
-        throw olfe
-      }
+      offenderEventQueueService.handleLockingFailure(olfe, RequeueDestination.OFFENDER, requestJson)
```

Comment:
Refactored into a method as it is now needed in both the offender and domain listeners